src/backend/distributed/deparser/qualify_aggregate_stmts.c

```diff
@@ -0,0 +1,34 @@
+/*-------------------------------------------------------------------------
+ *
+ * qualify_aggregate_stmts.c
+ *	  Functions specialized in fully qualifying all aggregate statements.
+ *    These functions are dispatched from qualify.c
+ *
+ *	  Fully qualifying aggregate statements consists of adding the schema name
+ *	  to the subject of the types as well as any other branch of the parsetree.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "catalog/namespace.h"
+#include "distributed/deparser.h"
+#include "nodes/makefuncs.h"
+#include "utils/lsyscache.h"
+
+void
+QualifyDefineAggregateStmt(Node *node)
+{
+	DefineStmt *stmt = castNode(DefineStmt, node);
+
+	if (list_length(stmt->defnames) == 1)
```

Comment:
Why do we check the length of list here? In fact I don't totally understand what we have as defnames, can you please explain?

Reply:
`defnames` contains the names of the aggregate to be created. If its length is 1, it means the schema name is not given. In that case we can simply qualify the node with the schema name